internal/cmd/all/list.go

```diff
@@ -0,0 +1,157 @@
+package all
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/hetznercloud/cli/internal/cmd/base"
+	"github.com/hetznercloud/cli/internal/cmd/certificate"
+	"github.com/hetznercloud/cli/internal/cmd/firewall"
+	"github.com/hetznercloud/cli/internal/cmd/floatingip"
+	"github.com/hetznercloud/cli/internal/cmd/image"
+	"github.com/hetznercloud/cli/internal/cmd/iso"
+	"github.com/hetznercloud/cli/internal/cmd/loadbalancer"
+	"github.com/hetznercloud/cli/internal/cmd/network"
+	"github.com/hetznercloud/cli/internal/cmd/output"
+	"github.com/hetznercloud/cli/internal/cmd/placementgroup"
+	"github.com/hetznercloud/cli/internal/cmd/primaryip"
+	"github.com/hetznercloud/cli/internal/cmd/server"
+	"github.com/hetznercloud/cli/internal/cmd/sshkey"
+	"github.com/hetznercloud/cli/internal/cmd/volume"
+	"github.com/hetznercloud/cli/internal/hcapi2"
+	"github.com/hetznercloud/cli/internal/state"
+	"github.com/hetznercloud/hcloud-go/v2/hcloud"
+)
+
+var listCmd = base.Cmd{
+	BaseCobraCommand: func(client hcapi2.Client) *cobra.Command {
+		cmd := &cobra.Command{
+			Use:   "list FLAGS",
+			Short: "List all resources",
+		}
+
+		cmd.Flags().StringP("selector", "l", "", "Selector to filter by labels")
+
+		cmd.Flags().Bool("paid", false, "Only list paid resources")
+
+		output.AddFlag(cmd, output.OptionJSON())
+
+		return cmd
+	},
+	Run: func(ctx context.Context, client hcapi2.Client, actionWaiter state.ActionWaiter, cmd *cobra.Command, args []string) error {
+
+		paid, _ := cmd.Flags().GetBool("paid")
+
+		outOpts := output.FlagsForCommand(cmd)
+
+		var cmds []base.ListCmd
+		if paid {
+			cmds = []base.ListCmd{
+				server.ListCmd,
+				loadbalancer.ListCmd,
+				primaryip.ListCmd,
+				floatingip.ListCmd,
+				image.ListCmd,
+				volume.ListCmd,
+			}
+		} else {
+			cmds = []base.ListCmd{
+				server.ListCmd,
+				network.ListCmd,
+				loadbalancer.ListCmd,
+				certificate.ListCmd,
+				firewall.ListCmd,
+				primaryip.ListCmd,
+				floatingip.ListCmd,
+				image.ListCmd,
+				volume.ListCmd,
+				iso.ListCmd,
+				placementgroup.ListCmd,
+				sshkey.ListCmd,
+			}
+		}
+
+		type response struct {
+			result []any
+			err    error
+		}
+
+		responseChs := make([]chan response, len(cmds))
+
+		for i, lc := range cmds {
+			i, lc := i, lc
+			ch := make(chan response)
+			responseChs[i] = ch
+
+			go func() {
+				defer close(ch)
+
+				labelSelector, _ := cmd.Flags().GetString("selector")
+				listOpts := hcloud.ListOpts{
+					LabelSelector: labelSelector,
+				}
+
+				flagSet := pflag.NewFlagSet(lc.JSONKeyGetByName, pflag.ExitOnError)
+
+				switch lc.JSONKeyGetByName {
+				case image.ListCmd.JSONKeyGetByName:
+					flagSet.StringSlice("type", []string{"backup", "snapshot"}, "")
+				case iso.ListCmd.JSONKeyGetByName:
+					flagSet.StringSlice("type", []string{"private"}, "")
+				}
+
+				_ = flagSet.Parse([]string{})
```

Comment:
Can you add a comment why this is necessary? To the unknowing reader this looks like a no-op.